src/vanilla/ClientModelExtensions.cs

```diff
@@ -346,7 +347,8 @@ public static string CheckNull(string valueReference, string executionBlock)
             if (sequence != null && sequence.ShouldValidateChain())
             {
                 var elementVar = scope.GetUniqueName("element");
-                var innerValidation = sequence.ElementType.ValidateType(scope, elementVar, null);
+                variable.Name = elementVar;
+                var innerValidation = sequence.ElementType.ValidateType(scope, variable, null);

```

Comment:
Why are you passing `variable` instead of `elementVar` here?

Reply:
Because I have changed  `string valueReference` to `IVariable variable` in ValidateType method [here](https://github.com/Azure/autorest.csharp/blob/3948893c945f85597fdebe3843d4183ce9360dfd/src/vanilla/ClientModelExtensions.cs#L322).

Reply:
https://github.com/Azure/autorest.csharp/pull/895#discussion_r488332308